### ISSUE TITLE
Solution hinting

### ISSUE THREAD

**Issue by Dimosts** [chocoteam/pychoco#13]
I saw that solution hinting is supported in JAVA.

Is it possible to have this also in the python API?

Currently adding Choco solver in CPMpy modeling library for python, and as CPMpy supports solution hinting, it would be nice to also have it with Choco.

**Comment by dimitri-justeau** [comment]
Hi Dimosts,

Very happy to know that you are adding Choco in CPMpy through our pychoco binding :-)

I did know about solution hinting, its a feature that I missed from Choco... But it should be definitely possible to include it in pychoco ! I won't be able to do it before one week or two, but as soon as I can I'll look at it !

Do you (or @cprudhom) know if there is any paper describing solution hinting ? I searched rapidly and did not find anything.

Cheers,

Dimitri.

**Comment by Dimosts** [comment]
I do not think there is any paper describing it. Based on what I see from OR-Tools, I think it basically starts the search for the specified variables with the specified values in the hint, before considering other values. There is also a discussion for this in https://github.com/google/or-tools/discussions/3277. 

I did think that it was implemented in Choco, it is mentioned in this release: https://github.com/chocoteam/choco-solver/releases

It is an optional thing in CPMpy, and nice to have, but not mandatory. Can just update it in CPMpy Choco interface when is included. Thanks.
